cmd/common/config.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/config"
+	apihelpers "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api-helpers"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+var configPath string
+
+// TODO: Perhaps this could be params so we can use the same mechanism for both
+var rootFlags = map[string][]string{
+	"gadgetctl":      {"verbose"},
+	"ig":             {"verbose", "auto-sd-unit-restart", "auto-mount-filesystems", "auto-wsl-workaround"},
+	"kubectl-gadget": {"verbose", "request-timeout", "client-certificate", "client-key", "insecure-skip-tls-verify", "kubeconfig", "server", "tls-server-name", "token"},
+}
```

Comment:
I was wondering why we need it and why not to use `VisitAll`, I suppose the reason is that operators / runtimes will also register global flags and we want to handle them in  SetFlagsForParams.

Reply:
I just realized we can still do that. I just needed to create a copy of flagset before we add runtimes / operators flags so we can use it afterwards. Updated the PR!